### PR TITLE
chore: standardise licenses on MIT

### PR DESCRIPTION
**Pull request by bradzacher**
## PR Checklist

- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Most of our licenses were MIT already.
This PR relicenses the BSD packages as MIT to standardise our licensing.
For standardisation I added the license to internal packages as well.

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
Whoops sorry, meant to get to this

File: packages/parser/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "url": "https://github.com/typescript-eslint/typescript-eslint/issues"
   },
   "homepage": "https://typescript-eslint.io/packages/parser",
-  "license": "BSD-2-Clause",
+  "license": "MITClause",
```
Comment (ericcornelissen) [inline packages/parser/package.json:31]: @JoshuaKGoldberg @bradzacher this looks like a typo:

```diff
-  "license": "MITClause",
+  "license": "MIT",
```

Comment (voxpelli) [reply packages/parser/package.json:31]: PR to fix this: https://github.com/typescript-eslint/typescript-eslint/pull/10481

**Comment by voxpelli** [comment]
A random note: While BSD-2-Clause code can be re-licensed as MIT it does so with one limitation: The original BSD-2-Clause license needs to be kept as well:

> - Redistributions of source code must retain the above copyright
>  notice, this list of conditions and the following disclaimer.

So just a FYI – this re-licensing is not strictly correct done, in case you care about it.

**Pull request merged**